libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -107,6 +107,52 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::copyToMemoryLiteralFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_memory_" + h256(_literal, h256::AlignLeft).hex();
```

Comment:
This does not compute a hash and will lead to a collision for strings that are the same in the first 256 bits.

Reply:
Do we have some helper function to calculate hash, or can I use `std::hash`?

Reply:
I used same method for calculating `richIdentifier` in `StringLiteralType`. Should be fine now I guess